### ISSUE TITLE
Regression: Fix support for descendants and variants

### ISSUE THREAD

**Issue by stevespringett** [CycloneDX/cyclonedx-core-java#75]
The move to Jackson in #69 broke support for descendants and variants in a components pedigree.

The code as of today specifies:
```java
@JacksonXmlElementWrapper(localName = "ancestors")
@JacksonXmlProperty(localName = "component")
public List<Component> getAncestors() {
    return ancestors;
}
```

However, when adding the following to
```java
@JacksonXmlElementWrapper(localName = "descendants")
@JacksonXmlProperty(localName = "component")
public List<Component> getDescendants() {
    return descendants;
}
```

breaks with the following exception:
`
org.cyclonedx.exception.ParseException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Conflicting getter definitions for property "component": org.cyclonedx.model.Pedigree#getAncestors() vs org.cyclonedx.model.Pedigree#getDescendants()
`

Jackson was removed previously in https://github.com/CycloneDX/cyclonedx-core-java/commit/92f9251076645221f60cc3adf35f7f646b15af66 due to this issue. A workaround will need to be implemented, even if it relies on an alternative framework.

See also: https://github.com/FasterXML/jackson-dataformat-xml/issues/192

**Comment by DarthHater** [comment]
@stevespringett you can go ahead and assign me to this! Gonna take a crack today!

**Comment by stevespringett** [comment]
Resolved with #76
